test/e2e-common.sh

```diff
@@ -89,10 +89,8 @@ function validate_run() {
 function run_yaml_tests() {
   echo ">> Starting tests"
 
-  find ${REPO_ROOT_DIR}/examples/ -name *.yaml -exec cat {} \; \
-      | sed 's/christiewilson-catfactory/${KO_DOCKER_REPO}/' \
-      | ko apply -f - \
-      || return 1
+  sed -i.bak 's/christiewilson-catfactory/${KO_DOCKER_REPO}/' examples/resources.yaml	 
```

Comment:
You're corrupting the original yaml file again. I'm strongly against such approach.

Reply:
The find statement was not applying all the files so the tests were not actually running, and I wasn't able to make it work. If you have any examples of something similar working, I can try to change it back and fix it.
The file is changed, but never pushed anywhere.

Reply:
Are we on different platforms? Because:

```shell
knative/build-pipeline$ REPO_ROOT_DIR=$PWD ; find ${REPO_ROOT_DIR}/examples/ -name *.yaml -exec ls {} \;
/usr/local/google/home/adrcunha/go/src/github.com/knative/build-pipeline/examples/run/task-run.yaml
/usr/local/google/home/adrcunha/go/src/github.com/knative/build-pipeline/examples/run/output-pipeline-run.yaml
/usr/local/google/home/adrcunha/go/src/github.com/knative/build-pipeline/examples/run/pipeline-run.yaml
/usr/local/google/home/adrcunha/go/src/github.com/knative/build-pipeline/examples/default-cluster-admin.yaml
/usr/local/google/home/adrcunha/go/src/github.com/knative/build-pipeline/examples/output-tasks.yaml
/usr/local/google/home/adrcunha/go/src/github.com/knative/build-pipeline/examples/pipeline.yaml
/usr/local/google/home/adrcunha/go/src/github.com/knative/build-pipeline/examples/deploy-task.yaml
/usr/local/google/home/adrcunha/go/src/github.com/knative/build-pipeline/examples/build-task.yaml
/usr/local/google/home/adrcunha/go/src/github.com/knative/build-pipeline/examples/resources.yaml
/usr/local/google/home/adrcunha/go/src/github.com/knative/build-pipeline/examples/output-pipeline.yaml
```

I know that the file is not pushed, but it's part of the repo and can be easily committed by mistake.

Reply:
And also:

```shell
build-pipeline$ REPO_ROOT_DIR=$PWD ; find ${REPO_ROOT_DIR}/examples/ -name resources.yaml -exec cat {} \; | sed 's/christiewilson-catfactory/${KO_DOCKER_REPO}/' > examples/resources.yaml_nocat
build-pipeline$ diff examples/resources.yaml examples/resources.yaml_nocat 
21c21
<     value: gcr.io/christiewilson-catfactory/leeroy-web
---
>     value: gcr.io/${KO_DOCKER_REPO}/leeroy-web
31c31
<     value: gcr.io/christiewilson-catfactory/leeroy-app
---
>     value: gcr.io/${KO_DOCKER_REPO}/leeroy-app
```

Reply:
The `find` lists the files correctly, but because its doing `cat` on all of them, there are not `---` separator between the different yaml definitions which results in errors when trying to `ko apply` 

I'll try to figure out a fix without changing the file

Reply:
Thanks for finding the issue. You can change the find command into a loop, like:

```shell
for file in $(find ${REPO_ROOT_DIR}/examples/ -name *.yaml); do
  sed 's/christiewilson-catfactory/${KO_DOCKER_REPO}/' ${file} | ko apply -f - || return 1
done
```

Reply:
Thanks :) will apply this change